### PR TITLE
Allow polars as valid output type

### PR DESCRIPTION
**Pull request by psmyth94**
I was trying out polars as an output for a map function and found that it wasn't a valid return type in `validate_function_output`. Thought that we should accommodate this by creating and adding it to the `allowed_processed_input_types` variable.

### CODE REVIEW

**Review by lhoestq — COMMENTED** [review]
Cool ! Sorry for the late review. Can you add a test to make sure map() with a polars-output function works ?
You can place the test in test_arrow_dataset.py

**Comment by psmyth94** [comment]
Hello @lhoestq, I added the test and modified [update_data](https://github.com/huggingface/datasets/blob/bececdac927160b5c7e883736d7cc79d5699ad0a/src/datasets/arrow_dataset.py#L3437) to include `polars` as an updatable type. Although, it seems pretty redundant to do the type checks both before `validate_function_output` and then immediately afterward within the call stack. Could consider adding `allowable_types` in `validation_function_output`.

File: tests/test_arrow_dataset.py [hunk 1]
```diff
@@ -1528,6 +1535,47 @@ def func_return_multi_row_pd_dataframe(x):
             with self._create_dummy_dataset(in_memory, tmp_dir) as dset:
                 self.assertRaises(ValueError, dset.map, func_return_multi_row_pd_dataframe)
 
+    def test_map_return_pl_dataframe(self, in_memory):
```
Comment (lhoestq) [inline tests/test_arrow_dataset.py:1538]:
```suggestion
    @require_polars
    def test_map_return_pl_dataframe(self, in_memory):
```

**Review by lhoestq — APPROVED** [review]
Cool ! thanks for the additions :)

It looks all good to me as is. Not a big deal to have it at two places, since it's not for the same objective (check whether the type is a data update type, and whether the type is a valid type)

**Pull request merged**